### ISSUE TITLE
[BUG] Initiating self-referencing documents with nested links breaks due to uncaught request loop in 1.16.0

### ISSUE THREAD

**Issue by TLeitzbach** [roman-right/beanie#449]
**Describe the bug**
The recursive check_nested_links() introduced in 1.16.0, beanie\odm\utils\init.py leads to a request loop resulting in an eventual break for self-referencing (linked) documents .

**To Reproduce**
Python 3.9

requirements.txt
beanie==1.16.0
pydantic==1.9.2
pymongo[srv]==4.1.1

```python
class Example(Document):
    custom_id: int
    example_name: str
    similar_examples: list[Link[Example]]


asyncio.run(init_beanie(..., document_models=[Example]))
```

Traceback (most recent call last):
  File "<input>", line 9, in <module>
  Python39\lib\asyncio\runners.py", line 44, in run
    return loop.run_until_complete(main)
  Python39\lib\asyncio\base_events.py", line 642, in run_until_complete
    return future.result()
 ...\am_mdb_utils.py", line 20, in init_mongo_client
    await init_beanie(database=mongoClient.client[mongoClient.databaseInfo["database"]],
 venv\lib\site-packages\beanie\odm\utils\init.py", line 475, in init_beanie
    await Initializer(
 venv\lib\site-packages\beanie\odm\utils\init.py", line 88, in __await__
    yield from self.init_class(model).__await__()
 venv\lib\site-packages\beanie\odm\utils\init.py", line 445, in init_class
    await self.init_document(cls)
 venv\lib\site-packages\beanie\odm\utils\init.py", line 345, in init_document
    self.init_document_fields(cls)
 venv\lib\site-packages\beanie\odm\utils\init.py", line 191, in init_document_fields
    check_nested_links(link_info)
 venv\lib\site-packages\beanie\odm\utils\init.py", line 180, in check_nested_links
    check_nested_links(nested_link_info)
 venv\lib\site-packages\beanie\odm\utils\init.py", line 180, in check_nested_links
    check_nested_links(nested_link_info)
 venv\lib\site-packages\beanie\odm\utils\init.py", line 180, in check_nested_links
    check_nested_links(nested_link_info)
  [Previous line repeated 957 more times]
 venv\lib\site-packages\beanie\odm\utils\init.py", line 173, in check_nested_links
    nested_link_info = detect_link(v)
 venv\lib\site-packages\beanie\odm\utils\relations.py", line 47, in detect_link
    return LinkInfo(
  File "pydantic\main.py", line 340, in pydantic.main.BaseModel.__init__
  File "pydantic\main.py", line 1076, in pydantic.main.validate_model
  File "pydantic\fields.py", line 884, in pydantic.fields.ModelField.validate
  File "pydantic\fields.py", line 1101, in pydantic.fields.ModelField._validate_singleton
  File "pydantic\fields.py", line 1148, in pydantic.fields.ModelField._apply_validators
  File "pydantic\class_validators.py", line 318, in pydantic.class_validators._generic_validator_basic.lambda13
  File "pydantic\validators.py", line 560, in pydantic.validators.make_class_validator.class_validator
  File "pydantic\utils.py", line 184, in pydantic.utils.lenient_issubclass
  Python39\lib\abc.py", line 123, in __subclasscheck__
    return _abc_subclasscheck(cls, subclass)
RecursionError: maximum recursion depth exceeded in comparison

**Expected behavior**
I am not entirely sure how this case should be handled forward-looking but it should not break the initialization. Perhaps a fall-back to the first instance Link as in previous versions would be good for now in this case. E.g. in beanie\odm\utils\init.py, line 190 ff.:

```python

 if link_info is not None:
       cls._link_fields[v.name] = link_info
        if k != link_info.field:
             check_nested_links(link_info)
```

**Additional context**
Thanks for your support! Love the package:)

**Comment by roman-right** [comment]
Should be fixed in `1.16.4`. Please, try

**Comment by TLeitzbach** [comment]
Hi @roman-right. I do not encounter the issue anymore. Thanks!!!
